java/src/test/resources/source/do_not_format_me/UnnecessaryModifier/NestingRecord_Issue846_Pre.java

```diff
@@ -22,4 +22,10 @@ public abstract class SomeAbstractInnerClass {
 
 	public static abstract @interface SomeAnnotation {
 	}
-}
\ No newline at end of file
+
+	public static enum SomeEnum {
+	}
+
+	public static final record SomeRecord(int i) {
+	}
```

Comment:
This example parses, but doesn't compile. Shall we fix that

```suggestion
	public static final record SomeNestedRecord(int i) {
	}
```

Reply:
Yes